omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/ConceptFragmentController.java

```diff
@@ -0,0 +1,41 @@
+package org.openmrs.module.registrationapp.fragment.controller.field;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import org.openmrs.Concept;
+import org.openmrs.ConceptAnswer;
+import org.openmrs.api.context.Context;
+import org.openmrs.ui.framework.annotation.FragmentParam;
+import org.openmrs.ui.framework.fragment.FragmentModel;
+import org.openmrs.util.LocaleUtility;
+
+public class ConceptFragmentController {
+
+	public void controller(FragmentModel model, @FragmentParam(required = true, value = "conceptId") Integer conceptId) throws Exception {
+		
+		Concept concept = Context.getConceptService().getConcept(conceptId);
+		if(concept == null){
+			model.put("conceptAnswers", new ArrayList<Map<String, String>>());
+		} else{
+			Collection<Map<String, String>> collection = new ArrayList<Map<String, String>>();
+			Map<String, String> conceptMap;
+			Locale locale = LocaleUtility.getDefaultLocale();
+			
+			Collection<ConceptAnswer> conceptAnswers = concept.getAnswers();
+			
+			for (ConceptAnswer conceptAnswer : conceptAnswers) {
+				conceptMap = new HashMap<String, String>();
+				String fullname = conceptAnswer.getAnswerConcept().getFullySpecifiedName(locale).getName();
+				conceptMap.put("label", fullname);
+				conceptMap.put("value", fullname);
```

Comment:
Should value be the uuid or id? Preferably uuid

Reply:
The thing is that person attributes are written to the database as a text value not a uuid. In the submit method when registering a patient, it tries to get person attributes from a config file that is defined in Manage Apps section of OpenMRS. And creates an instance of PersonAttribute class using a PersonAttributeType object and a text value of a personAttribute. And this code, this method works the same for all person attributes. I've found it right setting the fullname to the value. The method below that works currently in registrationapp module shows what i'm explaining
```
public static void resolvePersonAttributeFields(NavigableFormStructure form, Person person,
													Map<String, String[]> parameterMap) {
		List<Field> fields = form.getFields();
		if (fields != null && fields.size() > 0) {
			for (Field field : fields) {
				if (StringUtils.equals(field.getType(), "personAttribute")) {
					String[] parameterValues = parameterMap.get(field.getFormFieldName());
					if (parameterValues != null && parameterValues.length > 0) {
						if (parameterValues.length > 1) {
							log.warn("Multiple values for a single person attribute type not supported, ignoring extra values");
						}
						String parameterValue = parameterValues[0];
						if (parameterValue != null) {
							PersonAttributeType personAttributeByUuid = Context.getPersonService()
									.getPersonAttributeTypeByUuid(field.getUuid());
							if (personAttributeByUuid != null) {
								PersonAttribute attribute = new PersonAttribute(personAttributeByUuid, parameterValue);
								person.addAttribute(attribute);
							}
						}
					}
				}
			}
		}
	}
```